### PR TITLE
JSON Schema validation

### PR DESCRIPTION
**Pull request by aaronshiel**
JSON schema validations in place for API entry points that receive a payload.

### CODE REVIEW

**Review by kycarr — DISMISSED** [review]

File: mentor_upload_api/src/mentor_upload_api/blueprints/upload/answer.py [hunk 1]
```diff
@@ -64,13 +67,29 @@ def begin_tasks_in_parallel(req):
     return my_chord.delay()
 
 
+trim_existing_upload_json_schema = {
+    "type": "object",
+    "properties": {
+        "mentor": {"type": "string"},
```
Comment (mbsimonovic) [inline mentor_upload_api/src/mentor_upload_api/blueprints/upload/answer.py:73]: think you can add length requirements (min, max)

File: mentor_upload_api/src/mentor_upload_api/blueprints/upload/answer.py [hunk 2]
```diff
@@ -64,13 +67,29 @@ def begin_tasks_in_parallel(req):
...
+
+
 @answer_blueprint.route("/trim_existing_upload/", methods=["POST"])
 @answer_blueprint.route("/trim_existing_upload", methods=["POST"])
 def trim_existing_upload():
     req_log.info("trim existing, body: [%s]", request.form.get("body"))
     body = json.loads(request.form.get("body", "{}"))
     if not body:
         raise Exception("missing required param body")
+    validate_json(body, trim_existing_upload_json_schema)
```
Comment (mbsimonovic) [inline mentor_upload_api/src/mentor_upload_api/blueprints/upload/answer.py:92]: can you validate the `request.form.get("body", "{}")` before parsing to json?

**Comment by mbsimonovic** [comment]
nice 👏🏻 for a follow up, figure out how to do validation declaratively, using decorators or higher order functions or whatever:
```
# instead of this:
def handler(req):
  body = json.loads(req)
  validate_json(body, handler_json_schema)

# something like this:

@validate(schema=handler_json_schema)
def handler(req):
  body = json.loads(req)


``

File: mentor_upload_api/src/mentor_upload_api/helpers.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+def validate_json(json_schema):
+    def validate_json_wrapper(f):
+        @wraps(f)
+        def json_validated_function(*args, **kwargs):
+            try:
+                if not json_schema:
+                    raise Exception("'json_schema' param not provided to validator")
+                body = request.form.get("body", {})
+                if body:
+                    json_body = json.loads(body)
```
Comment (mbsimonovic) [inline mentor_upload_api/src/mentor_upload_api/helpers.py:24]: perhaps you should check Content-Type header before trying to parse?

**Pull request closed without merging**